tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -107,11 +107,11 @@ void ROOT::Experimental::Internal::RPageSinkBuf::UpdateSchema(const RNTupleModel
       auto p = &(*cloned);
       auto &projectedFields = changeset.fModel.GetProjectedFields();
       RNTupleModel::RProjectedFields::FieldMap_t fieldMap;
-      fieldMap[p] = projectedFields.GetSourceField(field);
+      fieldMap[p] = fInnerModel->FindField(projectedFields.GetSourceField(field)->GetQualifiedFieldName());
       auto targetIt = cloned->begin();
       for (auto &f : *field)
-         fieldMap[&(*targetIt++)] = projectedFields.GetSourceField(&f);
-      const_cast<RNTupleModel::RProjectedFields &>(fInnerModel->GetProjectedFields()).Add(std::move(cloned), fieldMap);
+         fieldMap[&(*targetIt++)] = fInnerModel->FindField(projectedFields.GetSourceField(&f)->GetQualifiedFieldName());
+      fInnerModel->fProjectedFields->Add(std::move(cloned), fieldMap);
```

Comment:
It has been uncovered by the unit test modification in this PR. It's a tricky one to pin down because the previous logic is fundamentally wrong, mixing fields from different models.

That said, I'm thinking that a "column hell" test would be good. A test that combines
  - A model with a nested hierarchy of fixed-size and variable length arrays
  - Projected regular fields
  - Projected extended fields
  - Regular extended fields
  - The last two in the same and different clusters and different cluster groups
  - Multiple column representations for those columns
  - Extended column representations

That requires some thinking and may be for a different PR though.

Reply:
Reordered and edited the commits such that unit tests correspond better to the fixes.